hypothesis-python/src/hypothesis/strategies/_internal/core.py

```diff
@@ -958,14 +959,13 @@ def builds(
             from hypothesis.strategies._internal.types import _global_type_lookup
 
             for kw, t in infer_for.items():
-                if (
-                    getattr(t, "__module__", None) in ("builtins", "typing")
```

Comment:
This example didn't actually work now, https://github.com/HypothesisWorks/hypothesis/issues/3026#issuecomment-981241222, because `Optional[RecursiveClass].__module__` is `typing`. `(RecursiveClass|None).__module__` is `types`, so that happened to work. And `builtins` is probably also unnecessary because relevant types are registered in `_global_type_lookup` anyway.

Long story short: Unless there's a reason I don't see, just remove the `__module__` check and everything is fine.

Reply:
Just FYI, this was the cause of the ghostwriter failures (now fixed).